### PR TITLE
relocate FsPath from stdlib::os and rename PyPathLike to OsPath

### PR DESCRIPTION
**Pull request by youknowone**
FsPath work to support #4773 

cc @ilp-sys 

`PyPathLike` is defined in `posixmodule` as `path_t`.
But FsPath is more common concept.

`PyPathLike` also is not a python object. We don't prefix non-python object with `Py` anymore without good reasons

### CODE REVIEW

File: vm/src/function/fspath.rs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    #[cfg(unix)]
+    pub fn to_cstring(&self, vm: &VirtualMachine) -> PyResult<std::ffi::CString> {
+        std::ffi::CString::new(self.as_bytes().to_vec()).map_err(|e| e.into_pyexception(vm))
+    }
+
+    #[cfg(windows)]
+    pub fn to_widecstring(&self, vm: &VirtualMachine) -> PyResult<widestring::WideCString> {
+        widestring::WideCString::from_os_str(&self.as_os_str()?)
```
Comment (fanninpm) [inline vm/src/function/fspath.rs:99]:
```suggestion
        widestring::WideCString::from_os_str(&self.as_os_str(vm)?)
```

**Comment by youknowone** [comment]
@fanninpm could you review it again?

File: common/src/os.rs [hunk 2]
```diff
@@ -21,3 +21,19 @@ pub fn errno() -> io::Error {
...
+#[cfg(unix)]
+pub fn bytes_as_osstr(b: &[u8]) -> Result<&std::ffi::OsStr, Utf8Error> {
+    use std::os::unix::ffi::OsStrExt;
+    Ok(std::ffi::OsStr::from_bytes(b))
+}
+
+#[cfg(not(unix))]
+pub fn bytes_as_osstr(b: &[u8]) -> Result<&std::ffi::OsStr, Utf8Error> {
+    Ok(std::str::from_utf8(b)?.as_ref())
+}
```
Comment (fanninpm) [inline common/src/os.rs:34]: Does this work?
```suggestion
pub fn bytes_as_osstr(b: &[u8]) -> Result<&std::ffi::OsStr, Utf8Error> {
    #[cfg(unix)]
    {
        use std::os::unix::ffi::OsStrExt;
        Ok(std::ffi::OsStr::from_bytes(b))
    }
    #[cfg(not(unix))]
    {
        Ok(std::str::from_utf8(b)?.as_ref())
    }
}
```

**Review by DimitrisJim — APPROVED** [review]

File: common/src/os.rs (hunk 2 above)
Comment (youknowone) [reply common/src/os.rs:34]: I prefer to keep separated functions. Those functions are small enough. But when the logic goes longer, single function commonly leads to `#[cfg]` spaghetti here and their while separated functions leads to create a new shared function.

**Pull request merged**